src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -293,6 +293,11 @@ def relative_to_exec_root(path):
     if not self.get_options().colors:
       zinc_args.append('-no-color')
 
+    if self.post_compile_extra_resources(ctx):
+      self.extra_post_compile_resources_digest(ctx)
```

Comment:
This is being called twice in some cases, which is odd. One approach would be to continue to write into an anonymous location, and then:
1. for hermetic, never materialize it to the absolute path: just merge it into the input snapshot
2. for non-hermetic, materialize it to the absolute path

Reply:
Yep sound good. That's what I meant by refactoring it a bit earlier as well :)

Reply:
Done.